hw06/ListManipulator 14.1/Form1.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using System.IO;
+
+namespace ListManipulator_14._1
+{
+    public partial class Form1 : Form
+    {
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void Form1_Load(object sender, EventArgs e)
+        {
+            List<int> numbers = new List<int>();
+
+            string[] lines = File.ReadAllLines("random.txt");
```

Comment:
The professor told us that our code should be "bullet proof".
I don't have a random.txt file in my computer which will cause an error. 
I think we should give the user an option to choose a file or find a way to keep the random.txt file in our project. 

//open text file
OpenFileDialog openFile = new OpenFileDialog()
{
    Filter = "Text Documents | *.txt",
    Multiselect = false,
    Title = "Textfiles only!"
};
if (openFile.ShowDialog() == DialogResult.OK)
{
    file_path = openFile.FileName;
    if (Validate_File(file_path))
    {
        path_text.Text = Path.GetFileName(file_path);
        Store_Value(file_path);
    }
    else
    {

    }
}
This is a logic that I use to open files from local. After it opens the txt file it will check the file if it is in the right format throught validate_file.
If you need validate_file, its in hw04 driver's license exam.

Reply:
As I said in my notes the random.txt file was located in the bin folder. However, I have now added it to out repo and re-implemented the code so that it is callable from other computers as well. We do not need to ask the user what file they want as the programming question 14.1 asks us to use the random.txt file from the student sample program of our textbook. Your code would not have solved the issue either. Your code only validates the file that is inputted by the user but that is not what the question wanted.